Encryption.md

```diff
@@ -158,12 +158,32 @@ the encryption buffer is comprised of an IV, ciphertext and tag, as described in
 Algorithms section. The length of the encryption buffer (a 4-byte little endian) is 
 written to the output stream, followed by the buffer itself.
 
+|length (4 bytes) | IV (12 bytes) | ciphertext (`length` bytes) | tag (16 bytes) |
+|-----------------|---------------|-----------------------------|----------------|
+
 The column pages (data and dictionary) are encrypted after compression. For each page, 
 the encryption buffer is comprised of an IV (nonce), ciphertext and (in case of AES_GCM_V1) of a 
 tag, as described in the Algorithms section. Only the buffer is written to the output 
 stream - no need to write the length of the buffer, since the length (page size after
 compression and encryption) is kept in the page headers.
 
+| IV (12 bytes) | ciphertext (`page_size` bytes) | tag (16 bytes) |
+|---------------|--------------------------------|----------------|
+
+or 
+
+| IV (16 bytes) | ciphertext (`page_size` bytes) | 
+|---------------|--------------------------------|
+
+Sometimes, a number of encrypted columns have their file offset either directly visible or 
+possible to infer from other metadata. This might be exploited for an exotic
+attack on file integrity (such as swapping dictionary pages of two columns). While these
+attacks succeed under rare conditions and in any case don't harm data confidentiality - 
+they can be prevented by using the offset as a part of the GCM AAD formation. Therefore,
```

Comment:
The offset of the very first page of every Parquet file is the same. Does this mean that a malicious person can swap the first pages of two Parquet files without knowing the encryption key?

Reply:
Nope, this won't work, but another attack might: replacement of column pages in row group 0 by pages from the same column in row group 1.